File: 0387-first-unique-character-in-a-string/0387-first-unique-character-in-a-string.py
class Solution:
    def firstUniqChar(self, s: str) -> int:
        dictionary=dict()
        for i in s:
            if i in dictionary:
                dictionary[i]+=1
            else:
                dictionary[i]=1
                
        for i in dictionary:
            if dictionary[i]==1:
                return s.index(i)
        return -1
        

File: 0387-first-unique-character-in-a-string/0387-first-unique-character-in-a-string.java
import java.util.HashMap;

public class Solution {
    public int firstUniqChar(String s) {
        HashMap<Character, Integer> dictionary = new HashMap<>();
        
        // Count the frequency of each character
        for (char c : s.toCharArray()) {
            dictionary.put(c, dictionary.getOrDefault(c, 0) + 1);
        }
        
        // Find the first unique character
        for (int i = 0; i < s.length(); i++) {
            if (dictionary.get(s.charAt(i)) == 1) {
                return i;
            }
        }
        
        return -1;
    }
}
